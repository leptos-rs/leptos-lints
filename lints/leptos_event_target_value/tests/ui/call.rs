use leptos::prelude;
use leptos::prelude::event_target_value;

fn main() {
    event_target_value::<u32>();

    let _a: Option<&str> = prelude::event_target_value();

    leptos::prelude::event_target_value::<String>();

    let _b: Option<Option<u8>> = ::leptos::prelude::event_target_value();
}
