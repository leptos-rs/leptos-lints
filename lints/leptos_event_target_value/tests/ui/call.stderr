warning: use of `leptos::prelude::event_target_value`
  --> $DIR/call.rs:5:5
   |
LL |     event_target_value::<u32>();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: this function can panic at run-time if the target doesn't have an event. For further information visit https://github.com/leptos-rs/leptos-lints/tree/main/lints/leptos_event_target_value#readme
   = note: `#[warn(leptos_event_target_value)]` on by default

warning: use of `leptos::prelude::event_target_value`
  --> $DIR/call.rs:7:28
   |
LL |     let _a: Option<&str> = prelude::event_target_value();
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: this function can panic at run-time if the target doesn't have an event. For further information visit https://github.com/leptos-rs/leptos-lints/tree/main/lints/leptos_event_target_value#readme

warning: use of `leptos::prelude::event_target_value`
  --> $DIR/call.rs:9:5
   |
LL |     leptos::prelude::event_target_value::<String>();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: this function can panic at run-time if the target doesn't have an event. For further information visit https://github.com/leptos-rs/leptos-lints/tree/main/lints/leptos_event_target_value#readme

warning: use of `leptos::prelude::event_target_value`
  --> $DIR/call.rs:11:34
   |
LL |     let _b: Option<Option<u8>> = ::leptos::prelude::event_target_value();
   |                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: this function can panic at run-time if the target doesn't have an event. For further information visit https://github.com/leptos-rs/leptos-lints/tree/main/lints/leptos_event_target_value#readme

warning: 4 warnings emitted

